### PR TITLE
feat:user side errors handler for login and register pages

### PR DESCRIPTION
**Pull request by FBagdeli**

### CODE REVIEW

**Review by MyrtheDullaart — DISMISSED** [review]

**Review by PerikK — DISMISSED** [review]

File: src/context/auth.jsx [hunk 1]
```diff
@@ -165,4 +187,24 @@ const ProtectedRoute = ({ children, disabledNav = false }) => {
     )
 }
 
+function validationEmail(email) {
```
Comment (vherus) [inline src/context/auth.jsx:190]: Let's rename this to be a verb: `validateEmail`

File: src/context/auth.jsx [hunk 2]
```diff
@@ -165,4 +187,24 @@ const ProtectedRoute = ({ children, disabledNav = false }) => {
...
+    const emailPattern = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+  
+    return !email || !emailPattern.test(email) ? false : true;
+  }
+  
+  function validationPassword(password) {
```
Comment (vherus) [inline src/context/auth.jsx:196]: `validatePassword`

File: src/context/auth.jsx [hunk 3]
```diff
@@ -165,4 +187,24 @@ const ProtectedRoute = ({ children, disabledNav = false }) => {
...
+    const hasNumber = /\d/.test(password);
+    const hasSpecialCharacter = /[!@#$%^&*(),.?":{}|<>]/.test(password);
+  
+    return password.length < minLength ||
+      !hasUppercase ||
+      !hasNumber ||
+      !hasSpecialCharacter
+      ? false
+      : true;
+  }
```
Comment (vherus) [inline src/context/auth.jsx:208]: All of these conditions will result in a true or false, you don't need to use a ternary to translate them aslong as you use the correct condition checks

File: src/service/errors.js [hunk 4]
```diff
@@ -1,4 +1,8 @@
 export default {
-    LOGIN_FAILED : 'An error occurred during login',
-    REGISTRATION_FAILED: 'An error occurred during registration',
-}
\ No newline at end of file
+  LOGIN_FAILED: "An error occurred during login",
+  REGISTRATION_FAILED: "An error occurred during registration",
+  PASSWORD_REQUIRMENTS:
+    "Password must be at least 8 characters long and include at least one uppercase letter, one number, and one special character.",
```
Comment (vherus) [inline src/service/errors.js:5]: The string should be on the same line as the key

File: src/context/auth.jsx [hunk 5]
```diff
@@ -75,6 +86,17 @@ const AuthProvider = ({ children }) => {
...
+                throw new Error(ERR.ENTER_EMAIL_PASSWORD);
+              }
+              const validatedEmail = validationEmail(email);
+              if (!validatedEmail) {
+                throw new Error(ERR.EMAIL_ERROR_MESSAGE);
+              }
+              const validatedPassword = validationPassword(password);
+              if (!validatedPassword) {
+                throw new Error(ERR.PASSWORD_REQUIRMENTS);
+              }
```
Comment (vherus) [inline src/context/auth.jsx:99]: I don't think we need to store these in variables

```js
if (!validateEmail(email))

if (!validatePassword(password))
```

File: src/context/auth.jsx [hunk 6]
```diff
@@ -44,6 +44,17 @@ const AuthProvider = ({ children }) => {
...
```
Comment (vherus) [inline src/context/auth.jsx:57]: I don't think we need to store these in variables

```js
if (!validateEmail(email))

if (!validatePassword(password))
```

**Review by vherus — CHANGES_REQUESTED** [review]

File: src/context/auth.jsx [hunk 7]
```diff
@@ -1,168 +1,210 @@
...
+  const handleRegister = async (email, password) => {
+    try {
+      if (!email || !password) {
+        throw new Error(ERR.ENTER_EMAIL_PASSWORD);
+      }
+      if (!validateEmail(email)) {
+        throw new Error(ERR.EMAIL_ERROR_MESSAGE);
+      }
+      if (!validatePassword(password)) {
+        console.log(!validatePassword(password))
```
Comment (vherus) [inline src/context/auth.jsx:94]: Let's get rid of this log

File: src/context/auth.jsx [hunk 8]
```diff
@@ -1,168 +1,210 @@
...
+
+  if (
+    password.length < minLength ||
+    !hasUppercase ||
+    !hasNumber ||
+    !hasSpecialCharacter
+  ) {
+    return false;
+  }
+  return true;
```
Comment (vherus) [inline src/context/auth.jsx:207]: If you flip the conditional checks, you can just return the result of the evaluation:

```js
return password.length >= minLength && hasUppercase && hasNumber && hasSpecialCharacter
```

**Review by vherus — CHANGES_REQUESTED** [review]

**Review by vherus — APPROVED** [review]

**Review by PerikK — APPROVED** [review]

**Pull request merged**